ci/action.yml

```diff
@@ -0,0 +1,149 @@
+          sudo rm /usr/local/bin/tfenv || echo "INFO: No tfenv installed"
+          sudo rm /usr/local/bin/terraform || echo "INFO: No terraform installed"
+          sudo ln -s ~/.tfenv/bin/* /usr/local/bin > /dev/null
+        fi
+    - name: Terraform Install
+      working-directory: ${{ inputs.tfpath }}
+      shell: bash
+      run: |
+        echo "****** INSTALL TERRAFORM ******"
+
+        if [ -f .terraform-version ]; then
+          tfenv install "$(cat .terraform-version)" > /dev/null
+        else
+          tfenv install 1.0.0 && tfenv use 1.0.0 > /dev/null
+        fi
+    - name: Check README.md
+      if: inputs.readme_check == 'true'
+      shell: bash
+      working-directory: ${{ inputs.tfpath }}
+      run: |
+        echo "****** CHECK README.MD ******"
+
+        if [[ ! -f README.md ]]; then
+          echo "ERROR: ${{ inputs.tfpath }}/README.md missing."
+          exit 1
+        fi
+    - name: Check .terraform-version
+      shell: bash
+      working-directory: ${{ inputs.tfpath }}
+      run: |
+        echo "****** CHECK .terraform-version ******"
+
+        if [[ ! -f .terraform-version ]]; then
+          echo "WARNING: ${{ inputs.tfpath }}/.terraform-version missing."
+        fi
+    - name: Check .terraform.lock.hcl
+      shell: bash
+      working-directory: ${{ inputs.tfpath }}
+      run: |
+        echo "****** CHECK .terraform.lock.hcl ******"
+
+        if [[ ! -f .terraform.lock.hcl ]]; then
+          echo "WARNING: ${{ inputs.tfpath }}/.terraform.lock.hcl missing."
+        fi
+    - name: Check terraform.required_version
+      shell: bash
+      working-directory: ${{ inputs.tfpath }}
+      run: |
+        echo "****** CHECK terraform.required_version ******"
+
+        match="[[:space:]]+terraform[[:space:]]+{[[:space:]]+required_version[[:space:]]+.*"
+        file_content=$(cat versions.tf)
+        
+        if [[ " $file_content " =~ $match ]]; then
+          echo "ERROR: ${{ inputs.tfpath }}/versions.tf missing `terraform.required_version`."
+          exit 1 
+        fi
+    - name: Project name matches terraform.backend.gcs.prefix
+      shell: bash
+      working-directory: ${{ inputs.tfpath }}
+      run: |
+        echo "****** PROJECT NAME MATCHES terraform.backend.gcs.prefix ******"
+
+        if [[ ! -f terraform.tf ]]; then
+          echo "INFO: ${{ inputs.tfpath }}/.terraform.tf not present."
+        else
+          directory=$(echo ${{ inputs.tfpath }} | cut -d'/' -f1 -s)
+          file_content=$(cat terraform.tf)
+          json_content=$(echo $file_content | json2hcl --reverse)
+
+          if echo $json_content | jq -re 'has("terraform")' > /dev/null; then
+            if echo $json_content | jq -re '.terraform[] | has("backend")' > /dev/null; then
+              if echo $json_content | jq -re '.terraform[].backend[] | has("gcs")' > /dev/null; then
+                gcs_prefix=$(echo $json_content | jq -re '.terraform[].backend[].gcs[].prefix')
```

Comment:
Do we need these conditionals to check the structure of terraform.tf? jq in `gcs_prefix=$(echo $json_content | jq -re '.terraform[].backend[].gcs[].prefix')` will return a non-zero exit code if it can't parse the filter.

Reply:
i can add the shorter jq - honestly, this conditional structure just came about as i was parsing the data and looking for other errors with the hcl2json library & whitespaces (which popped up elsewhere).

Reply:
changed